istio-interop-testing/src/main/java/io/grpc/testing/istio/EchoTestServer.java

```diff
@@ -158,20 +206,31 @@ public static void main(String[] args) throws Exception {
     echoTestServer.blockUntilShutdown();
   }
 
-  void runServers(List<Integer> grpcPorts, String hostname) throws IOException {
+  void runServers(String hostname, List<Integer> grpcPorts, List<Integer> xdsPorts,
+      List<Integer> tlsPorts, String forwardingAddress, ServerCredentials tlsServerCredentials)
+      throws IOException {
     ServerServiceDefinition service = ServerInterceptors.intercept(
-        new EchoTestServiceImpl(hostname), new EchoTestServerInterceptor());
+        new EchoTestServiceImpl(hostname, forwardingAddress), new EchoTestServerInterceptor());
     servers = new ArrayList<>(grpcPorts.size() + 1);
     for (int port : grpcPorts) {
-      runServer(port, service);
+      ServerCredentials serverCredentials = InsecureServerCredentials.create();
+      if (xdsPorts.contains(port)) {
+        serverCredentials = XdsServerCredentials.create(InsecureServerCredentials.create());
+      } else if (tlsPorts.contains(port)) {
+        serverCredentials = tlsServerCredentials;
+      }
+      servers.add(runServer(port, service, serverCredentials));
     }
   }
 
-  void runServer(int port, ServerServiceDefinition service) throws IOException {
+  static Server runServer(
+      int port, ServerServiceDefinition service, ServerCredentials serverCredentials)
+      throws IOException {
     logger.log(Level.INFO, "Listening GRPC on " + port);
-    servers.add(Grpc.newServerBuilderForPort(port, InsecureServerCredentials.create())
+    return Grpc.newServerBuilderForPort(port, serverCredentials)
         .addService(service)
-        .build().start());
+        .build()
```

Comment:
I have done this but I think this will cause problems. When EXPOSE_GRPC_ADMIN env var is set, both the grpc go echo server and the grpc java echo server will create admin services at the same port numbers and because they both  run in the same pod this will cause port conflicts. I will test and then remove the code if needed